api/api/auth/views.py

```diff
@@ -71,3 +73,20 @@ def _find_or_create_user(self, user_info):
                 first_name=user_info.get('first_name'),
                 last_name=user_info.get('last_name'))
         return user
+
+
+class DeleteTokenView(APIView):
+    """View to delete the tokens."""
+
+    def delete(self, request):
+        serializer = DeleteTokenRequestSerializer(data=request.data)
```

Comment:
Now that I think about it, we don't even need a `token` in the request `body`. If the user is authenticated, you can access the `user` from `request.user`.

Then use `related_name` to access it's token.

```python
user_token = request.user.token
if token:
    token.delete()
```

We dont need to throw an error if token doesn't exist. We can simply consider it deleted and return `HTTP_204`.

Reply:
> 

If the user is authenticated then the token will always be valid, isn't it?

Reply:
Think about this case: User logs in using two devices.

Reply:
Ok so from our model implementation of `Token` a user can have multiple `tokens` . When we use `request.user.token` to `delete` the token how would we know which of the tokens to be deleted? Or we are aiming to logout the `user` from all the devices.

`Why a user can have multiple tokens ?`
AFAIK since we haven't specified `primary key=true` on `user` attribute in `Token model` due to which `django` will treat the `id` column as `primary key` so a `user` can have multiple tokens

Reply:
`user` is a `OneToOneField` in `Token` model:
https://github.com/BitByte-TPC/gymkhana/blob/4fb305f8ac76daf21b7321618b3f233704414f9e/api/api/auth/models.py#L10